torch/distributed/pipelining/_backward.py

```diff
@@ -1,12 +1,234 @@
 # mypy: allow-untyped-defs
 # Copyright (c) Meta Platforms, Inc. and affiliates
-from typing import List, Optional
+import collections
+import logging
+import weakref
+from typing import Any, cast, Deque, Dict, Iterator, List, Optional, Set, Tuple, Union
 
 import torch
+from torch.autograd.graph import GradientEdge, Node
+from torch.nn import Parameter
 
 from ._debug import map_debug_info
 
 
+logger = logging.getLogger(__name__)
+
+
+def _get_grad_fn_or_grad_acc(t: torch.Tensor) -> Union[Node, None]:
+    if t.requires_grad and t.grad_fn is None:
```

Comment:
can we have comments about the overall behavior of this fn? why does takign a view get us a grad_fn when there is none? what does none mean? which users would fall into the RuntimeError path?

cc @soulitzer

Reply:
In the requires_grad=True and grad_fn is None path, since accumulate grad node is lazily created, we need to create a dummy view in order to trigger its creation. grad_fn of the view can be None if we are in a no-grad context.

This function as-is can also return None if t.requries_grad is False.

Reply:
Added comments, thanks Jeffrey!